emukit/quadrature/methods/vanilla_bq.py

```diff
@@ -4,48 +4,67 @@
 
 import numpy as np
 from scipy.linalg import lapack
+from typing import Tuple
 
 from .warped_bq_model import WarpedBayesianQuadratureModel
+from emukit.quadrature.interfaces.base_gp import IBaseGaussianProcess
 
 
 class VanillaBayesianQuadrature(WarpedBayesianQuadratureModel):
     """
     class for vanilla Bayesian quadrature
     """
 
-    def __init__(self, base_gp):
+    def __init__(self, base_gp: IBaseGaussianProcess):
         """
         :param base_gp: a model derived from BaseGaussianProcess
         """
         super(VanillaBayesianQuadrature, self).__init__(base_gp)
 
-    def transform(self, Y):
+    def transform(self, Y: np.ndarray) -> np.ndarray:
         """ Transform from base-GP to integrand """
         return Y
 
-    def inverse_transform(self, Y):
+    def inverse_transform(self, Y: np.ndarray) -> np.ndarray:
         """ Transform from integrand to base-GP """
         return Y
 
-    def predict_base(self, X_pred, return_full_cov=False):
-        """ Computes the predictive mean and covariance """
-        m, cov = self.base_gp.predict(X_pred, return_full_cov)
+    def predict_base(self, X_pred: np.ndarray) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
```

Comment:
Wander if that should be a private method

Reply:
I thought about that, too, then I opted for this because some people might want to e.g. plot the base GP. Otherwise one would need to call `loop.model.base_gp.predict()` instead of `loop.model.predict_base()`.
But I am also fine with changing it, whatever you think is a better user experience.